uk.ac.kcl.inf.modelling.xdsml.gemoc_henshin/src/uk/ac/kcl/inf/modelling/xdsml/gemoc_henshin/engine/core/HenshinStep.xtend

```diff
@@ -0,0 +1,86 @@
+package uk.ac.kcl.inf.modelling.xdsml.gemoc_henshin.engine.core
+
+import java.util.List
+import org.eclipse.emf.ecore.EObject
+import org.eclipse.emf.ecore.EcoreFactory
+import org.eclipse.emf.henshin.interpreter.Match
+import org.eclipse.gemoc.trace.commons.model.generictrace.impl.GenericSmallStepImpl
+import org.eclipse.gemoc.trace.commons.model.trace.TracePackage
+/**
+ * a class representing one step of execution
+ */
+class HenshinStep extends GenericSmallStepImpl {
+	
+	protected Match match
+	protected List<Match> matches
+	
+	/**
+	 * create a new HenshinStep with a match
+	 * @param match
+	 */
+	new(Match match) {
+		super()
+		this.match = match
+	}
+	/**
+	 * create a new Henshin step with a sequence of rule matches
+	 * @param a list of matches
+	 */
+	new(List<Match> matches) {
+		super()
+		this.matches = matches
+	}
+	
+	/**
+	 * return a MSEOccurence for a step
+	 */
+	override getMseoccurrence() {
+		if(matches === null || matches.isEmpty){
+			generateMSE(match, match.getRule().getName(), match.toString())
+		}else{
+			var rulesNames = ''
+			for(Match m: matches){
+				var r = m.getRule();
+				rulesNames = rulesNames + ' ' + r.getName()
+			}
+			generateMSE(matches.get(0), rulesNames, rulesNames)
+		}
+	}
+	/**
+	 * mock the generation of MSEOccurences
```

Comment:
This may need a little more explanation: why is it necessary to mock these? what are MSEOccurences in GEMOC (maybe add a link to relevant documentation)?